packages/jupyter-chat/src/model.ts

```diff
@@ -520,6 +553,53 @@ export class ChatModel implements IChatModel {
    */
   inputChanged?(input?: string): void {}
 
+  /**
+   * Add attachment to send with next message.
+   */
+  addAttachment = (attachment: IAttachment, emit: boolean = true): void => {
+    const index = this.inputAttachments.findIndex(
+      att => att.type === attachment.type && att.value === attachment.value
+    );
+    if (index > -1) {
+      this.inputAttachments.splice(index, 1, attachment);
+    } else {
+      this.inputAttachments.push(attachment);
+    }
+    if (emit) {
+      this._inputAttachmentsChanges.emit([...this.inputAttachments]);
+    }
+  };
```

Comment:
1. Is there a need for an `emit` argument if we usually default to `true`? IMO, allowing consumers to not emit to `inputAttachmentsChanges` defeats the purpose of also allowing consumers to listen to that signal.


2. Can this logic be simplified further? The ordering of attachments should not matter, so we can return early if a duplicate attachment already exists. For example:


```suggestion
  addAttachment = (attachment: IAttachment): void => {
    const duplicateAttachment = this.inputAttachments.find(
      att => att.type === attachment.type && att.value === attachment.value
    );
    if (duplicateAttachment) {
      return;
    }
    
    this.inputAttachments.push(attachment);
    this._inputAttachmentsChanges.emit([...this.inputAttachments]);
  };
```